richtextfx/src/integrationTest/java/org/fxmisc/richtext/style/StylingTests.java

```diff
@@ -0,0 +1,160 @@
+package org.fxmisc.richtext.style;
+
+import static org.junit.Assert.assertEquals;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import org.fxmisc.flowless.Cell;
+import org.fxmisc.flowless.VirtualFlow;
+import org.fxmisc.richtext.InlineCssTextAreaAppTest;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+import com.nitorcreations.junit.runners.NestedRunner;
+
+import javafx.scene.Node;
+import javafx.scene.layout.Region;
+import javafx.scene.shape.Path;
+import javafx.scene.text.Text;
+import javafx.scene.text.TextFlow;
+
+@RunWith(NestedRunner.class)
+public class StylingTests extends InlineCssTextAreaAppTest {
+
+    private final static String firstWord = "Hello ";
+    private final static String styledWord1 = "World";
+    private final static String moreText = " and also the ";
+    private final static String styledWord2 = "Sun";
+    private final static String remainingLine = " and Moon";
+
+    
+    /**
+     * 
+     * @param index The index of the desired paragraph box
+     * @return The paragraph box for the paragraph at the specified index
+     */
+    private Region getParagraphBox(int index) {
+        @SuppressWarnings("unchecked")
+        VirtualFlow<String, Cell<String, Node>> flow = (VirtualFlow<String, Cell<String, Node>>) area.getChildrenUnmodifiable().get(index);
+        Cell<String, Node> gsa = flow.getCell(0);
+
+        // get the ParagraphBox (protected subclass of Region) 
+        return (Region) gsa.getNode();
+    }
+    
+
+    /**
+     * @return A list of text nodes which render the current text.
+     */
+    private List<Text> getTextNodes() {
+        // get the ParagraphBox (protected subclass of Region) 
+        Region paragraphBox = getParagraphBox(0);
+
+        // get the ParagraphText (protected subclass of TextFlow) 
+        TextFlow tf = (TextFlow) paragraphBox.getChildrenUnmodifiable().get(0);
```

Comment:
Could this be changed to check for the instance first, only cast if it's valid and throw an exception if it's not? Otherwise, I am not sure it will be the first one if a selection is made.

Reply:
True. get(0) is not always correct - I need to iterate over the children and look up the TextFlow (for which there should be only one, though),